### DISCUSSION TITLE
Tracking associations in JSON fields

### DISCUSSION THREAD

**Discussion by JoshuaBehrens** [shopware/platform#3148] [RFC / Proposal]
**Effort**: high

### Description: 

Slot configurations and custom fields are just some bunch of JSON. As soon as you start referencing entities in there you have a "dangling pointer" when the referenced entity is removed. There is no way via DAL nor SQL to track this nicely. It can happen easily to delete a media element that is used somewhere else and you won't know you broke something. There is a try to prevent this for medias but it is not working. The DAL can't handle this.

I think this is not a feature but a breaking change as with this support we can make a promise that this application will work correctly. But EVERYONE has to make sure they change their application to support this promise.

### Benefits

Less fragile custom fields and CMS elements.

### Break strategy

This probably means somewhere has to be stated which JSON sub field references to entities. Multiples? Single one?

**Comment by keulinho** [comment]
Thx @JoshuaBehrens for your proposal, but actually there was not much movement around this in the last year(s) and we don't expect that to change, therefore to be completely transparent we'll close this proposal for now.

**Reply by JoshuaBehrens** [reply]
Didn't you already change that for media entities? Can't we normalize this for every entity?
